cli.py

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env python
+"""
+This example uses docopt with the built in cmd module to demonstrate an
+interactive command application.
+Usage:
+    smart-library desk --action=<action> --name=<name> --i2c_address=<i2c_address>
+    smart-library (-i | --interactive)
+    smart-library (-h | --help | --version)
+Options:
+    -i, --interactive  Interactive Mode
+    -h, --help  Show this screen and exit.
+"""
+
+import sys, cmd, re
+from docopt import docopt, DocoptExit
+from services.desk_handler_service import DeskHandlerService
+from colorama import Fore, Style
+
+def docopt_cmd(func):
+    """
+    This decorator is used to simplify the try/except block and pass the result
+    of the docopt parsing to the called action.
+    """
+    def fn(self, arg):
+        try:
+            opt = docopt(fn.__doc__, self.format_args(arg))
+        except DocoptExit as e:
+            # The DocoptExit is thrown when the args do not match.
+            # We print a message to the user and the usage block.
+
+            print(Fore.RED + 'Whoops! Something went wrong. Please try again.')
+            print(e + Style.RESET_ALL)
```

Comment:
I get an exception here saying that e must be a string (but is of type DocoptExit). You can get to this statement by executing 'desk' on its own in the CLI

Reply:
Just do str(e) + Style.RESET_ALL